src/runtime/InternalLoader.js

```diff
@@ -158,7 +158,7 @@ class BundledCodeUnit extends CodeUnit {
     var normalizedNames =
         this.deps.map((name) => this.loader_.normalize(name));
     var module = this.execute.apply(Reflect.global, normalizedNames);
-    System.set(this.normalizedName, module);
+    $traceurRuntime.setModule(this.normalizedName, module);
```

Comment:
Currently we define `ModuleStore` in script that loads at the top of the `bin/traceur.js` file. `ModuleStore` defines the functions we use to store modules. If we 
```
import {setModule} from './ModuleStore.js';
```
then we have to
```
export let setModule = ModuleStore.set;
```
which will then fail when we compile as script,
```
/work/traceur-compiler/src/runtime/ModuleStore.js:304:3: Unexpected reserved word export
```
Maybe you know another way to accomplish this? If we switched completely to inline and gave up on `registerModule` then we could emit `ModuleStore` as a module, but in that case we don't need `registerModule` or `ModuleStore`.   

Alternatively we can split the binary into three parts, loader, runtime, and compiler.  The loader could be dependent upon the runtime and compiler; compiler could be dependent only on runtime. The different module loading formats would have different requirements in use.  For example, the inline format would not require the loader component.  This is the direction I was aiming for.